src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -853,10 +931,12 @@ private void AppendIntersectionWithPredecessor(IndentedStringBuilder.DelimitedSc
             {
                 delimited.BeginDelimitedElement();
 
-                string columnToSelect = (searchParamTableExpression.Kind == SearchParamTableExpressionKind.Chain ? searchParamTableExpression.ChainLevel - 1 : searchParamTableExpression.ChainLevel) == 0 ? "Sid1" : "Sid2";
+                bool intersectWithFirst = (searchParamTableExpression.Kind == SearchParamTableExpressionKind.Chain ? searchParamTableExpression.ChainLevel - 1 : searchParamTableExpression.ChainLevel) == 0;
 
-                StringBuilder.Append(VLatest.Resource.ResourceSurrogateId, tableAlias).Append(" IN (SELECT ").Append(columnToSelect)
-                    .Append(" FROM ").Append(TableExpressionName(predecessorIndex)).Append(")");
+                StringBuilder.Append("EXISTS(SELECT * FROM ").Append(TableExpressionName(predecessorIndex))
```

Comment:
>* [](http://example.com/codeflow?start=52&length=1)

I wonder, does SQL optimize this and check only where clause fields , or it's actually reads content of whole row just to perform exist on it....

Reply:
It optimizes. It will use the best index, which does not have to include all columns.